sdk/schemaregistry/Microsoft.Azure.Data.SchemaRegistry.ApacheAvro/src/SchemaRegistryAvroObjectSerializer.cs

```diff
@@ -0,0 +1,234 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Avro;
+using Avro.Generic;
+using Avro.IO;
+using Avro.Specific;
+using Azure.Core;
+using Azure.Core.Serialization;
+using Azure.Data.SchemaRegistry;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.Serialization;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+#pragma warning disable AZC0001 // Use one of the following pre-approved namespace groups (https://azure.github.io/azure-sdk/registered_namespaces.html): Azure.AI, Azure.Analytics, Azure.Data, Azure.DigitalTwins, Azure.Iot, Azure.Learn, Azure.Media, Azure.Management, Azure.Messaging, Azure.Search, Azure.Security, Azure.Storage, Azure.Template, Azure.Identity, Microsoft.Extensions.Azure
+namespace Microsoft.Azure.Data.SchemaRegistry.ApacheAvro
+#pragma warning restore AZC0001 // Use one of the following pre-approved namespace groups (https://azure.github.io/azure-sdk/registered_namespaces.html): Azure.AI, Azure.Analytics, Azure.Data, Azure.DigitalTwins, Azure.Iot, Azure.Learn, Azure.Media, Azure.Management, Azure.Messaging, Azure.Search, Azure.Security, Azure.Storage, Azure.Template, Azure.Identity, Microsoft.Extensions.Azure
+{
+    /// <summary>
+    /// A <see cref="SchemaRegistryAvroObjectSerializer"/> implementation that uses <see cref="SchemaRegistryClient"/> for Avro serialization/deserialization.
+    /// </summary>
+    public class SchemaRegistryAvroObjectSerializer : ObjectSerializer
+    {
+        private readonly SchemaRegistryClient _client;
+        private readonly string _groupName;
+        private readonly SchemaRegistryAvroObjectSerializerOptions _options;
+
+        /// <summary>
+        /// Initializes new instance of <see cref="SchemaRegistryAvroObjectSerializer"/>.
+        /// </summary>
+        public SchemaRegistryAvroObjectSerializer(SchemaRegistryClient client, string groupName, SchemaRegistryAvroObjectSerializerOptions options = null)
+        {
+            _client = client ?? throw new ArgumentNullException(nameof(client));
+            _groupName = groupName ?? throw new ArgumentNullException(nameof(groupName));
+            _options = options;
+        }
+
+        private static readonly byte[] EmptyRecordFormatIndicator = { 0, 0, 0, 0 };
+        private static readonly Encoding Utf8Encoding = new UTF8Encoding();
```

Comment:
This is where you can pass the param indicating whether to include BOM:
https://docs.microsoft.com/en-us/dotnet/api/system.text.utf8encoding.-ctor?view=netcore-3.1#System_Text_UTF8Encoding__ctor_System_Boolean_